### DISCUSSION TITLE
create range requests on mobiles devices

### DISCUSSION THREAD

**Discussion by dnltsk** [protomaps/PMTiles#173] [Q&A]
Mobile apps are often written in Kotlin for Android and Swift for iOS. Are there any libs around to support range requests on the client side? If not, is there a spec that explains the conversion from z/x/y coordinates to range requests?

In my case, I want to add vector PMTiles on an Android google maps client.

**Comment by dnltsk** [comment]
The [JS code](https://github.com/protomaps/PMTiles/blob/main/js/index.ts#L98) seem pretty self explanatory. Maybe that helps already.

**Comment by bdon** [comment]
(moving to discussion)

It depends on the client library. Of the popular ones I know of (MapLibre Native, etc) there isn't a plugin system for data sources so would require maintaining a fork of the library. Can you point to any related open source data layer support for Android on google maps?

**Comment by do-me** [comment]
Some folks and I have been exploring this for tauri-based mobile apps. Got it working here on iOS and Android: https://github.com/inro-digital/tauri-offline-maps/issues/2 (but afaik getting user's current geolocation wasn't working when I last checked a few months ago, so a no-go for my use-case unfortunately).
